File: Java/src/main/java/wavenet-tts-project/TTS_Client.java
package wavenet_tts_project;

// Imports the Google Cloud client library
import com.google.cloud.texttospeech.v1.AudioConfig;
import com.google.cloud.texttospeech.v1.AudioEncoding;
import com.google.cloud.texttospeech.v1.SsmlVoiceGender;
import com.google.cloud.texttospeech.v1.SynthesisInput;
import com.google.cloud.texttospeech.v1.SynthesizeSpeechResponse;
import com.google.cloud.texttospeech.v1.TextToSpeechClient;
import com.google.cloud.texttospeech.v1.VoiceSelectionParams;
import com.google.protobuf.ByteString;
import java.io.FileOutputStream;
import java.io.OutputStream;

public class TTS_Client {
    private void TTS_Client() {
        
    }

    public void Create_Client() throws Exception {
        // Credentials in $GOOGLE_APPLICATION_CREDENTIALS environment variable
        // Instantiates a client
        try (TextToSpeechClient textToSpeechClient = TextToSpeechClient.create()) {
            // Set the text input to be synthesized
            SynthesisInput input = SynthesisInput.newBuilder()
            .setText("This is a test in java.")
            .build();

            // Build the voice request and set the language codes
            VoiceSelectionParams voice = VoiceSelectionParams.newBuilder()
            .setLanguageCode("en-US")
            .setName("en-US-Wavenet-C")
            .build();

            // Select the type of audio file you want returned
            AudioConfig audioConfig = AudioConfig.newBuilder()
            .setAudioEncoding(AudioEncoding.MP3)
            .build();

            // Perform the TTS request on the text input with
            // the selected voice parameters
            SynthesizeSpeechResponse response = textToSpeechClient
            .synthesizeSpeech(input, voice, audioConfig);

            // Get the audio contents from the response
            ByteString audioContents = response.getAudioContent();

            // Write the response to the output file
            try (OutputStream out = new FileOutputStream("output.mp3")) {
                out.write(audioContents.toByteArray());
                System.out.println("Audio content written to file: output.mp3");
            }
        }


    }    
}

File: Python/TTS_Client.py
# Test implementation of Google's WaveNet TTS API

from google.cloud import texttospeech
import os

def getSpeech(message):
    # Import Google Service Credentials
    os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = 'Text to Speech Project-02d00975225a.json'

    # Instantiates a client
    client = texttospeech.TextToSpeechClient()

    # Set the input to be synthesized
    synthesis_input = texttospeech.types.SynthesisInput(text=message)

    # Build the voice request, select the language code ("en-US") and the
    # ssml voice generator ("neutral")
    voice = texttospeech.types.VoiceSelectionParams(
        language_code='en-US',
        name='en-US-Wavenet-C'
    )

    # Select the type of audio file we want returned
    audio_config = texttospeech.types.AudioConfig(
        audio_encoding=texttospeech.enums.AudioEncoding.MP3
    )

    # Perform the text to speech request on the text input with the selected
    # voice parameters and audio file type
    response = client.synthesize_speech(synthesis_input, voice, audio_config)

    # The response's audio_content is binary
    with open('output.mp3', 'wb') as out:
        # Write the response to the output file
        out.write(response.audio_content)
        print("Audio content written to: output.mp3")